guard/src/rules/functions/date_time_tests.rs

```diff
@@ -0,0 +1,103 @@
+use std::{convert::TryFrom, rc::Rc};
+
+use crate::rules::{
+    eval_context::eval_context_tests::BasicQueryTesting,
+    exprs::AccessQuery,
+    functions::date_time::{now, parse_epoch},
+    path_value::PathAwareValue,
+    EvalContext, QueryResult,
+};
+use chrono::Utc;
+use pretty_assertions::assert_eq;
+use rstest::rstest;
+
+const VALUE_STR: &str = r#"
+{
+    "Resources": {
+        "LambdaFunction": {
+            "Type": "AWS::Lambda::Function",
+            "Properties": {
+                "UpdatedAt": "2024-08-21T00:00:00Z",
+                "CreatedAt": "2024-08-13T00:00:00Z",
+                "BadValue": "not-a-date"
+            }
+        }
+    }
+}
+    "#;
+
+#[rstest]
+#[case(
+    r#"Resources[ Type == 'AWS::Lambda::Function' ].Properties.UpdatedAt"#,
```

Comment:
nit: might be easier to read if we just format the last part of the query. 

ex 

```rust
let root_query = format!(r#"Resources[ Type == 'AWS::Lambda::Function' ].Properties.{query_str}"#, ); 
```
this would allow us to have the `query_str` just be CreatedAt/BadValue/etc..

```